src/file/FileUtils.js

```diff
@@ -426,29 +426,53 @@ define(function (require, exports, module) {
     }
     
     /**
+     * @private
+     * OS-specific helper for `compareFilenames()`
+     * @return {Function} The OS-specific compare function
+     */
+    var _cmpNames = (function () {
+        if (brackets.platform === "win") {
+            // Use this function on Windows
+            return function (filename1, filename2, lang) {
+                var f1 = getFilenameWithoutExtension(filename1),
+                    f2 = getFilenameWithoutExtension(filename2);
+                return f1.localeCompare(f2, lang, {numeric: true});
+            };
+        }
+        
+        // Use this function other OSes
+        return function (filename1, filename2, lang) {
+            return filename1.localeCompare(filename2, lang, {numeric: true});
+        };
+    }());
+    
+    /**
      * Compares 2 filenames in lowercases. In Windows it compares the names without the
      * extension first and then the extensions to fix issue #4409
      * @param {string} filename1
      * @param {string} filename2
      * @param {boolean} extFirst If true it compares the extensions first and then the file names.
-     * @return {number} The result of the local compare function
+     * @return {number} The result of the compare function
      */
     function compareFilenames(filename1, filename2, extFirst) {
-        var ext1   = getFileExtension(filename1),
-            ext2   = getFileExtension(filename2),
-            lang   = brackets.getLocale(),
-            cmpExt = ext1.toLocaleLowerCase().localeCompare(ext2.toLocaleLowerCase(), lang, {numeric: true}),
-            cmpNames;
+        var lang = brackets.getLocale();
```

Comment:
Would it be faster if lang could be a fourth optional parameter, and we can use it like `lang = lang || brackets.getLocale()`, so that it would be possible to call that function once, and not on every compare?